### PR TITLE
Add PyInstaller automation

### PR DESCRIPTION
**Pull request by jerbmega**
This PR adds a second workflow that automatically builds a release with PyInstaller, increments the patch number in semantic versioning by 1 based on the version number of the previous option, and uploads a release. This action will only run if CI passes, and if the target commit was pushed to main.

I also adjusted `win_build.py` to reflect that qwindows.dll and libEGL.dll are located in `Lib/site-packages/PyQt5/Qt5/...` now instead of `Lib/site-packages/PyQt5/Qt/...`

### CODE REVIEW

**Comment by Zamiell** [comment]
nice work

File: .github/workflows/pyinstaller_win.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Create a release archive
+        run: |
+          powershell Compress-Archive dist/* BasementRenovator.zip
+
+      - name: Upload release
+        uses: ncipollo/release-action@v1
+        with:
+          artifacts: "BasementRenovator.zip"
+          tag: ${{steps.versioning.outputs.version}}
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Zamiell) [inline .github/workflows/pyinstaller_win.yml:48]: ill put a newline here

**Comment by Zamiell** [comment]
k ill upload GITHUB_TOKEN as a secret

**Comment by jerbmega** [comment]
> k ill upload GITHUB_TOKEN as a secret

It's autogenerated for the action so you shouldn't need to do that, I tried that initially too but it doesn't let you make a secret with that name

**Comment by Zamiell** [comment]
OH

**Comment by Zamiell** [comment]
ok lets try merging this then AND SEE WHAT HAPPENS

**Pull request merged**